packages/tx/src/eip2930Transaction.ts

```diff
@@ -186,7 +205,17 @@ export default class AccessListEIP2930Transaction extends BaseTransaction<Access
     // Verify the access list format.
     AccessLists.verifyAccessList(this.accessList)
 
-    this.gasPrice = new BN(toBuffer(gasPrice === '' ? '0x' : gasPrice))
+    const gasPriceB = toBuffer(gasPrice === '' ? '0x' : gasPrice)
+    if (gasPriceB.length > 1 && gasPriceB[0] === 0x00) {
+      // RLP encoded integer values with leading zeroes are invalid
+      throw new Error('gasPrice cannot have leading zeroes')
```

Comment:
I am a bit confused, why this single extra check for this one parameter here?

Reply:
I think I just forgot to remove it when cleaning up after my last round of changes.  Will remove in the next commit.